src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -952,6 +954,38 @@ int TPCHandler::ProcessPullReq(const std::string &resource, XrdHttpExtReq &req)
             logTransferEvent(LogMask::Error, rec, "PULL_FAIL", msg);
             return req.SendSimpleResp(rec.status, NULL, NULL, msg, 0);
     }
+    // ddavila 2023-01-05:
+    // The following change was required by the Rucio/SENSE project where
+    // multiple IP addresses, each from a different subnet, are assigned to a
+    // single server and routed differently by SENSE.
+    // The above requires the server to utilize the same IP, that was used to
+    // start the TPC, for the resolution of the given TPC instead of
+    // using any of the IPs available.
+    if (!strcmp("fixed", m_route)){
```

Comment:
Hmmm, let's simplify this. See my comment above where you parse the value to make this a simple bool test.